mapping/react-leaflet.md

```diff
@@ -0,0 +1,433 @@
+
+### Render the GeoJSON
+Now the fun part — fetch and render the GeoJSON. You're going to do all of this in the `ChicagoWardMap` component. In `src/maps/wards.js`:
+```jsx
+import React, { useEffect, useState } from 'react'
+import BaseMap from './base'
+import { GeoJSON } from 'react-leaflet'
+
+function ChicagoWardMap() {
+    const [wardBorders, setWardBorders] = useState(null)
+
+    useEffect(() => {
+        // get the geojson
+        fetch('https://raw.githubusercontent.com/datamade/chicago-judicial-elections/master/wards/wards_2012.geojson')
+            .then((res) => res.json()) // parse the response into json
+            .then((geojson) => {
+                setWardBorders(geojson) // with the geojson, set the state for wardBorders
+            })
+    }, [setWardBorders])
+
+    const fill = {
+        fillColor: '#daf0ce', 
+        weight: 0.5,
+        opacity: 0.4,
+        color: '#666',
+        fillOpacity: 0.5
+    }
+
+    return (
+        <BaseMap center={[41.8781, -87.6298]} zoom={10}>
+                {/* this will only show when wardBorders has a truthy value */}
+                {wardBorders && <GeoJSON
+                                    key='ward-layer'
+                                    data={wardBorders}
+                                    style={fill} />}
+        </BaseMap>
+    )
+}
+
+export default ChicagoWardMap
+```
+
+This code:
+- Creates getters and setters for the `wardBorder` state:  `const [wardBorders, setWardBorders] = useState(null)`.
+- `useEffect` executes when the component renders.
+    - It uses `fetch` to retrieve the geojson.
+    - Parses the geojson and sets it as the value for `wardBorder`.
+- When `wardBorder` has a value, it will be passed into the `GeoJSON` component as `data`. This should show the Chicago Judicial Ward boundaries from 2012.
+
+Note that you don't have to fetch the GeoJSON. You can import the JSON directly from a file in your React app's code. This all depends on your application's architecture. This fetching pattern would allow for more dynamic mapping, like if you wanted a user to toggle between maps and your GeoJSON was retrievable from an API.
+
+## Attach click event callbacks to each GeoJSON feature
+Finally, we can attach callbacks to each GeoJSON feature. This is how you can add logic whenever a user clicks on a feature.
+
+You want to send a callback function into the GeoJSON layer. The GeoJSON component includes a prop called `onEachFeature`, where you can access the GeoJSON layer within the leaflet instance. Add this method inside the scope of the `ChicagoWardMap` component:
+```jsx
+function eventHandlersOnEachFeature(feature, layer) {
+    layer.on({
+      click: onWardClick
+    })
+}
+```
+This is an undocumented method in `react-leaflet`, but it's using the underlying `leaflet` library and HTML DOM events. See [the `leaflet` documentation](https://leafletjs.com/reference-1.7.1.html#domevent) for more details.
```

Comment:
Can you expand on this a bit? What's going on in this method? Is it safe to use an undocumented method?